.github/workflows/test.yaml

```diff
@@ -0,0 +1,59 @@
+name: Test
+on: [ push, pull_request, workflow_dispatch ]
+jobs:
+  unit:
+    runs-on: ubuntu-latest
+    name: Unit tests
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4.2.2
+      - name: Run tests
+        run: |
+          docker compose -f docker-compose.test.yaml up unit_test --abort-on-container-failure
+      - name: Archive code coverage results
+        uses: actions/upload-artifact@v4
+        with:
+          name: unit_coverage_${{ github.sha }}
+          path: ./coverage.out
+
+  integration:
+    name: Integration tests
+    runs-on: ubuntu-latest
+    needs: unit
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4.2.2
+      - name: Download coverage from unit tests
+        uses: actions/download-artifact@v4
+        with:
+          name: unit_coverage_${{ github.sha }}
+      - name: Run integration tests
+        run: |
+          docker compose -f docker-compose.test.yaml up integration_test --abort-on-container-failure
```

Comment:
nitpick: the command never cleans up the toxiproxy container (which is started in the background and keeps running). By changing this to:
```
docker compose -f docker-compose.test.yaml up toxiproxy integration_test --abort-on-container-exit
```
all other containers will be stopped once the first container exists, which is imo cleaner. Note that `abort-on-container-exit` is key here, as `abort-on-container-failure` would just hang.

Same reasoning for the `unit` job.